### PR TITLE
arch/imxrt1020: enable gpio/adc init in imxrt1020-evk

### PR DESCRIPTION
**Pull request by an4967**
- enable gpio/adc initialize in imxrt1020
- change gpio/adc conifg in imxrt1020/1050

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [562d7ba3d8e18c9efd0e66b45c56c0934f4256dd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c28ec26d048e4ab4c04259455b48d34fb2797c7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c28ec26d048e4ab4c04259455b48d34fb2797c7] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_adc.c:46: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:47: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:50: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:51: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:67: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:477: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:481: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:482: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:483: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:484: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:486: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:487: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:488: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:489: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:491: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:492: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:493: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:494: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:495: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:496: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.c:834: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.h:460: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [fc3a1b76bd5e5f9dbe3fd6a6589a6b873eb69086] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fc3a1b76bd5e5f9dbe3fd6a6589a6b873eb69086] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_adc.c:835: ERROR: [IDT_M_TAB] please, no space before tabs
os/arch/arm/src/imxrt/imxrt_adc.h:496: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [a3d0ab20d5d1f5b0fcaf629d8c4cafd5e34ae226] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a3d0ab20d5d1f5b0fcaf629d8c4cafd5e34ae226] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 1]
```diff
@@ -43,12 +43,12 @@
...
-#define IMXRT_ADC_DEFAULT_PEROID  10
-#define IMXRT_ADC_MIN_PEROID    1
-#define IMXRT_ADC_MAX_PEROID    1000000
+#define IMXRT_ADC_DEFAULT_PEROID	10
+#define IMXRT_ADC_MIN_PEROID		1
+#define IMXRT_ADC_MAX_PEROID		1000000
 
-#define CONVERT_ONECHANNEL		1
-#define ADC_CHANNEL_GROUP       0
+#define CONVERT_ONECHANNEL			1
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Please aline macro values

File: os/arch/arm/src/imxrt/imxrt_adc.c [hunk 2]
```diff
@@ -465,22 +466,13 @@ static void adc_conversion(int argc, uint32_t arg)
...
-	if (priv->current >= priv->nchannels) {
-		priv->current = 0;
-	}
-
-	/*next conversion */
-	wd_cancel(priv->work);
-	wd_start(priv->work, PeroidPerChannel, (wdentry_t)&adc_conversion, 1, (uint32_t)priv);
 #else
+	// TO DO
+#if 0
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.c]: Please remove this code if not needed.

File: os/arch/arm/src/imxrt/imxrt_adc.h [hunk 3]
```diff
@@ -62,9 +62,59 @@ extern "C" {
...
+#define IMXRT_ADC1_IN1	IOMUXC_GPIO_AD_B0_14_GPIO1_IO14
+#define IMXRT_ADC1_IN2	IOMUXC_GPIO_AD_B0_15_GPIO1_IO15
+#define IMXRT_ADC1_IN3	IOMUXC_GPIO_AD_B1_01_GPIO1_IO17
+#define IMXRT_ADC1_IN4	IOMUXC_GPIO_AD_B1_03_GPIO1_IO19
+#define IMXRT_ADC1_IN5	IOMUXC_GPIO_AD_B1_05_GPIO1_IO21
+#define IMXRT_ADC1_IN6	IOMUXC_GPIO_AD_B1_06_GPIO1_IO22
+#define IMXRT_ADC1_IN7	IOMUXC_GPIO_AD_B1_07_GPIO1_IO23
+#define IMXRT_ADC1_IN8	IOMUXC_GPIO_AD_B1_08_GPIO1_IO24
+#define IMXRT_ADC1_IN9	IOMUXC_GPIO_AD_B1_09_GPIO1_IO25
+#define IMXRT_ADC1_IN10	IOMUXC_GPIO_AD_B1_10_GPIO1_IO26
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.h]: Please aline macro.

File: os/arch/arm/src/imxrt/imxrt_adc.h [hunk 4]
```diff
@@ -62,9 +62,59 @@ extern "C" {
...
+#define IMXRT_ADC1_IN1	IOMUXC_GPIO_AD_B0_12_GPIO1_IO12
+#define IMXRT_ADC1_IN2	IOMUXC_GPIO_AD_B0_13_GPIO1_IO13
+#define IMXRT_ADC1_IN3	IOMUXC_GPIO_AD_B0_14_GPIO1_IO14
+#define IMXRT_ADC1_IN4	IOMUXC_GPIO_AD_B0_15_GPIO1_IO15
+#define IMXRT_ADC1_IN5	IOMUXC_GPIO_AD_B1_00_GPIO1_IO16
+#define IMXRT_ADC1_IN6	IOMUXC_GPIO_AD_B1_01_GPIO1_IO17
+#define IMXRT_ADC1_IN7	IOMUXC_GPIO_AD_B1_02_GPIO1_IO18
+#define IMXRT_ADC1_IN8	IOMUXC_GPIO_AD_B1_03_GPIO1_IO19
+#define IMXRT_ADC1_IN9	IOMUXC_GPIO_AD_B1_04_GPIO1_IO20
+#define IMXRT_ADC1_IN10	IOMUXC_GPIO_AD_B1_05_GPIO1_IO21
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_adc.h]: same as above.

File: os/board/imxrt1020-evk/src/imxrt_boot.c [hunk 5]
```diff
@@ -67,10 +67,105 @@
...
+			/* Register the ADC driver at "/dev/adc0" */
+			snprintf(path, sizeof(path), "/dev/adc%d", ADC_ASSIGN(bus, ch));
+
+			ret = adc_register(path, adc);
+			if (ret < 0) {
+				lldbg("ERROR: adc_register[%d] failed\n", ch);
+				return;
+			}
+		}
+
```
Comment (NilabhShant) [inline os/board/imxrt1020-evk/src/imxrt_boot.c]: Please remove extra line.

File: os/board/imxrt1050-evk/src/imxrt_boot.c [hunk 6]
```diff
@@ -118,18 +119,30 @@ void imxrt_board_adc_initialize(void)
...
+		imxrt_adc_pins_init(bus, sizeof(chanlist) / sizeof(uint8_t));
+
+		for (ch = 0; ch < sizeof(chanlist) / sizeof(uint8_t); ch++) {
+			adc = imxrt_adc_initialize(bus, chanlist[ch]);
+			if (NULL != adc) {
+				lldbg("VERVOSE: Success to get the imxrt ADC driver[%d]\n", ch);
+			} else {
+				lldbg("ERROR: up_spiinitialize failed\n");
+			}
+
```
Comment (NilabhShant) [inline os/board/imxrt1050-evk/src/imxrt_boot.c]: same as above.

**Comment by seinfra** [comment]
Target : [401b981f4a427443dd759ca44cb2bdcaed054f8a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [401b981f4a427443dd759ca44cb2bdcaed054f8a] - Code Rule Check (C++) OK.

**Comment by an4967** [comment]
@NilabhShant 
Thanks for your comments. I applied your mentions.

**Pull request merged**